backend/fms_core/resources/project_link_sample.py

```diff
@@ -0,0 +1,104 @@
+import reversion
+from import_export.fields import Field
+from import_export.widgets import ForeignKeyWidget
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
+from ._generic import GenericResource
+from ._utils import skip_rows, add_columns_to_preview
+from ..models import Project, Sample, Container, SampleByProject
+from ..utils import get_normalized_str
+
+
+__all__ = ["ProjectLinkSampleResource"]
+
+ADD_ACTION = "ADD"
+REMOVE_ACTION = "REMOVE"
+
+class ProjectLinkSampleResource(GenericResource):
+    action = Field(column_name='Action')
+    project_name = Field(column_name='Project Name') 
+    sample_name = Field(column_name='Sample Name')
+    barcode = Field(column_name='Sample Container Barcode',
+                     widget=ForeignKeyWidget(Container, 'barcode'))
+    coordinates = Field(column_name='Sample Container Coord')
+
+    class Meta:
+        model = SampleByProject
+        export_order = ('action', 'project_name', 'sample_name', 'barcode', 'coordinates',)
+
+    def before_import(self, dataset, using_transactions, dry_run, **kwargs):
+        skip_rows(dataset, 7)
+
+    def import_obj(self, obj, data, dry_run, **kwargs):
+        errors = {}
+        # Get field info
+        try:
+            super().import_obj(obj, data, dry_run, **kwargs)
+        except ValidationError as e:
+            errors.update(e.update_error_dict(errors).copy())
+        # Get sample object
+        try:
+            obj.sample = Sample.objects.get(name=data["Sample Name"],
```

Comment:
Are both fields (sample name and container information) necessary in the template? Is it still ok if we just get the Container info?

Reply:
I would either keep it required or just remove it altogether ... we do not really need it, but making it optional will just make it so no one fills it (making it useless). This is a small layer of validation, if it is too demanding for the lab we could remove it in a future iteration.

Reply:
Yeah, I wasnt sure what we decided on required fields. If the two are mandatory that's fine.